services/motion/builtin/move_request.go

```diff
@@ -805,3 +884,57 @@ func (mr *moveRequest) stop() error {
 	}
 	return nil
 }
+
+// getCurrentPosition returns a kinematic bases current position with respect to the world frame.
+func (mr *moveRequest) getCurrentPosition(ctx context.Context) (*referenceframe.PoseInFrame, error) {
+	currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// make sure the currentPosition is in the world frame
+	if currentPosition.Parent() != referenceframe.World {
+		// get the current inputs from the frame system service
+		currentInputs, _, err := mr.fsService.CurrentInputs(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		// transform using the planRequest's framesystem
+		tf, err := mr.planRequest.FrameSystem.Transform(currentInputs, currentPosition, referenceframe.World)
+		if err != nil {
+			return nil, err
+		}
+
+		// assert back to PoseInFrame
+		var ok bool
+		if currentPosition, ok = tf.(*referenceframe.PoseInFrame); !ok {
+			return nil, errors.New("cannot assert transformable as PoseInFrame")
+		}
+	}
+	return currentPosition, nil
+}
+
+// getExistingGeometries returns elements of the worlstate which are not transient.
+// Depending on the caller, the geometries returned are either in their relative position
+// with respect to the base or in their absolute position with respect to the world.
+func (mr *moveRequest) getExistingGeometries(
```

Comment:
This function doesn't make sense as currently written.  The worldstate should not include anything transient so I dont understand why this would ever be necessary.  You should be able to delete this and simply call `ObstaclesInWorldFrame` from the calling function.

Reply:
You're right worldstate does not include anything transient.
However, the way obstacles are digested for plan construction, `replan` and plan checking `checkPlan`.
replan asks for obstacles wrt the the base and checkplan asks for obstacles in the world frame.

Hence, when we know we will enter `CheckPlan` we need to edit what `ObstaclesInWorldFrame` returns since they will be with respect to the base frame but specified as with respect to the world frame.
Here, `f`, makes it such that values `ObstaclesInWorldFrame` may be put into world coordinates.